### PR TITLE
[trivial] Either make private or document undocumented public symbols in stdio

### PR DESCRIPTION
**Pull request by JackStouffer**

### CODE REVIEW

**Comment by 9il** [comment]
What the reason of that change? This will break existing code

File: std/stdio.d [hunk 1]
```diff
@@ -561,11 +564,12 @@ First calls $(D detach) (throwing on failure), and then attempts to
 associate the given Windows $(D HANDLE) with the $(D File). The mode must
 be compatible with the access attributes of the handle. Windows only.
 
-Throws: $(D ErrnoException) in case of error.
+Throws: $(D ErrgnoException) in case of error.
```
Comment (jmdavis) [inline std/stdio.d]: Somehow, I don't think that you meant to insert a `g` here.

File: std/stdio.d [hunk 2]
```diff
@@ -3872,7 +3878,7 @@ struct lines
             return opApplyRaw(dg);
         }
     }
-    // no UTF checking
+    /// Implements `opApply` `foreach` support but with no UTF checking
     int opApplyRaw(D)(scope D dg)
```
Comment (jmdavis) [inline std/stdio.d]: This looks like it's just intended to be used by `opApply` and should not be treated as public.

**Comment by jmdavis** [comment]
> What the reason of that change? This will break existing code

If something isn't documented, then it doesn't exist. Anyone who uses an undocumented symbol has no guarantees that their code won't be broken by changes to it later.

**Comment by 9il** [comment]
> If something isn't documented, then it doesn't exist. Anyone who uses an undocumented symbol has no guarantees that their code won't be broken by changes to it later.

I don't agree. No guarantees, but it still exists. And an engineer do not need to spend time copypast code

File: std/stdio.d (hunk 1 above)
Comment (JackStouffer) [inline std/stdio.d]: nope

File: std/stdio.d (hunk 2 above)
Comment (JackStouffer) [inline std/stdio.d]: ok, will mark as private

**Comment by JackStouffer** [comment]
These symbols were completely undocumented. The only way people were using these is if they found them by pure chance or they were looking at Phobos code. Either way, they aren't officially part of the Phobos API, so this is not a breaking change

**Comment by JackStouffer** [comment]
FIxed

**Comment by 9il** [comment]
The only real way to understand phobos is looking into its code. Out documentation is not so good

**Comment by jmdavis** [comment]
> The only real way to understand phobos is looking into its code. Out documentation is not so good

It's as good or better than most of the documentation for C++'s standard library, and you don't go looking though the STL and decide to use undocumented symbols do you?

**Comment by 9il** [comment]
> The only real way to understand phobos is looking into its code. Out documentation is not so good
> It's as good or better than most of the documentation for C++'s standard library, and you don't go looking though the STL and decide to use undocumented symbols do you?

I don't. STL has much more users and online activity. STL and Boost sources looks terrible for me. Phobos sources are readable and clean.

**Comment by jmdavis** [comment]
> I don't. STL has much more users and online activity. STL and Boost sources looks terrible for me. Phobos sources are readable and clean.

So, because there are fewer people to ask questions about Phobos, and it's actually readable, you think that folks should be able to expect to rely on undocumented symbols? I really don't think that that's reasonable. If they were intended to be part of the public API, they would be documented. The fact that they're public is either a mistake or a workaround for a problem and not because it's something that anyone should be using.

**Comment by jmdavis** [comment]
@JackStouffer It looks like you have some trailing whitespace, and the auto-tester doesn't like that.

**Comment by WalterBright** [comment]
I think that code was written by @andralex I think he should comment on whether they are meant to be public or not.

**Comment by WalterBright** [comment]
@JackStouffer Please in the future use more descriptive branch names than things like `package2`, which would be helpful when looking at https://auto-tester.puremagic.com/pulls.ghtml?projectid=1

This could be called `stdio-private`, for example. Something to jog the memory on what it is about.

**Comment by andralex** [comment]
There are two kinds of undocumented public symbols: (a) some that were inadvertent, (b) some that used to be documented and have been shunned by being left out there but undocumented in order to not break code.

I encourage making those in category (a) private, but (b)s should be left alone (at best with a comment indicating they're to stay put. @JackStouffer could you do some detective work on the file history to see which is which? Thx!

**Comment by JackStouffer** [comment]
Here's my philosophy:

1. If something is public, then it's meant to be used by the programmer.
2. If something is not meant to be used by the programmer, it should be private
2. If it's meant to be used by the programmer, then the programmer should know about it

I believe these changes fall in line with that. The structs were never meant to be used directly; you're supposed to use the factory functions.

**Comment by schuetzm** [comment]
I expect category (b) to have a (non-ddoc) comment stating that the symbols are there for legacy reasons.

**Comment by andralex** [comment]
Auto-merge toggled on

**Comment by BBasile** [comment]
breakage confirmed since 2.072. DFMT uses a few of these symbols. See https://github.com/Hackerpilot/dfmt/blob/master/src/dfmt/main.d#L118

**Comment by qchikara** [comment]
I suggest that these should be deprecated BEFORE private.
If we can make it private, we could also rename to _privateName and add a temporary alias orgName with the attr deprecated, then eventually omit such alias.

**Pull request merged**